### DISCUSSION TITLE
Noob Help : Setting up qtile

### DISCUSSION THREAD

**Discussion by manujchandra** [qtile/qtile#2735] [Q&A] [answered]
Hi,

I am new to qtile. I use Xonsh and Ranger though. I would like to integrate qtile with fedora xfce spin. Is this possible?

Is it possible to use xfce as the desktop environment (de) (or even gnome 40 for that matter would do) and qtile as the window manager (wm)? Could anyone guide me how to do this? Any links tor resources would also do. Because Fedora has a i3wm spin but not qtile.

Is qtile only a wm or can it also be used as a de+wm?

Thanks for the guidance.
Regards.

**Comment by m-col** [comment]
Hi there!

In theory it should be possible; Qtile can be used as a WM as part of another DE. In the docs there is a page for doing so with gnome: http://docs.qtile.org/en/latest/manual/config/gnome.html. Perhaps some of that page will also apply to xfce. If you can find instructions on how to run i3wm inside xfce, those same steps will very likely also work with qtile, so that might be your best bet.

Hope that helps!

**Reply by manujchandra** [reply]
Thanks. I will review this and try it on to see if I am able to make it work.

**Reply by manujchandra** [reply]
Hi,
I was able to follow [this tutorial](https://www.youtube.com/watch?v=rnZm3g_rbwM) and get it to install on a VM. 

Its giving a config error though and looks like it needs to be setup. I will google that separately.

Thanks for the help.

**Comment by manujchandra** [comment, marked as answer]
Hello,

Thanks for your help. I was able to get it working. Here are the steps I followed in case someone else needs to do it:

- Install Fedora 34 Xfce spin
- Install qtile latest version. dnf does not have latest version. Use `pip install qtile` (I got version 0.18)
- Press Alt+F3 and go to Window Manager > Keyboard. Clear all shortcuts.
- Press Alt+F3 and go to Keyboards and remove all shortcuts using the super key. Do not remove Alt+F3!
- Press Alt+F3 and go to Sessions and Startups, change "restart style" to _never_ for:
  xfwm4
  xfce4-panel : xfce panel also needs to be stopped from the startup applications, this will allow icons to appear in qtile.
  xfdesktop
- Press Save Session Button
- In Application Autostart tab add the command
`qtile start`
- Logout and Login

**Reply by nielsonrolim** [reply]
Thank you very much!

It worked on Manjaro XFCE too.

Just one comment: make sure there is no application opened when you click the `Save Session` button. Otherwise this application will open everytime you login.

**Reply by SweepingBishops** [reply]
How about adding this to the docs too?

**Reply by neerajadhav** [reply]
This is an amazing appraoch!  👍🏼

**Comment by manujchandra** [comment]
I have created a YouTube video on how to do this:

https://youtu.be/zdESlgVl0IU

Hope it will help others onboard QTile.
